Core/src/org/sleuthkit/autopsy/datamodel/AbstractContentNode.java

```diff
@@ -240,4 +280,79 @@ public int getContentChildrenCount() {
     public int read(byte[] buf, long offset, long len) throws TskException {
         return content.read(buf, offset, len);
     }
+    
+    
+    /**
+     * Updates the values of the properties in the current property sheet with
+     * the new properties being passed in. Only if that property exists in the
+     * current sheet will it be applied. That way, we allow for subclasses to
+     * add their own (or omit some!) properties and we will not accidentally
+     * disrupt their UI.
+     *
+     * Race condition if not synchronized. Only one update should be applied at
+     * a time.
+     *
+     * @param newProps New file property instances to be updated in the current
+     *                 sheet.
+     */
+    protected synchronized void updateSheet(NodeProperty<?>... newProps) {
+        //Refresh ONLY those properties in the sheet currently. Subclasses may have 
+        //only added a subset of our properties or their own props.s
+        Sheet visibleSheet = this.getSheet();
+        Sheet.Set visibleSheetSet = visibleSheet.get(Sheet.PROPERTIES);
+        Property<?>[] visibleProps = visibleSheetSet.getProperties();
+        for (NodeProperty<?> newProp : newProps) {
+            for (int i = 0; i < visibleProps.length; i++) {
+                if (visibleProps[i].getName().equals(newProp.getName())) {
+                    visibleProps[i] = newProp;
+                }
+            }
+        }
+        visibleSheetSet.put(visibleProps);
+        visibleSheet.put(visibleSheetSet);
+        //setSheet() will notify Netbeans to update this node in the UI.
+        this.setSheet(visibleSheet);
+    }
+    
+    /**
+     * Reads and returns a list of all tags associated with this content node.
+     * 
+     * @return list of tags associated with the node.
+     */ 
+    abstract protected List<Tag> getAllTagsFromDatabase();
```

Comment:
I went back and forth on this in my head several times - regarding pro and cons of defining abstract methods vs null/dummy implementation in the base class.  For exactly the same reason, I actually originally went with having null  methods in the base class so I wouldn't have to have duplicate them in the said classes - but then Codacy complained and suggested base class methods should be abstract,  so I switched it.  If you really feel strongly about it, I can go back to it.   Probably the 'pure' approach is to have have abstract methods in the base class, and define  an intermediate class with null implementations that both VolumeNode & ImageNode derive from, but I'm trying to avoid introducing yet another class in the hierarchy...

Reply:
I definitely do not feel strongly one way or another. If codacy wants it this way that is certainly a benefit.